src/main/java/core/basesyntax/service/strategy/FruitTransactionStrategyImpl.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class FruitTransactionStrategyImpl implements FruitTransactionStrategy {
+    private final Map<Operation, FruitTransactionHandler> map;
+
+    public FruitTransactionStrategyImpl(Map<Operation, FruitTransactionHandler> map) {
+        this.map = map;

```

Constructor does not validate the injected `map`. According to the task guidance and SOLID principles you should validate constructor parameters and fail fast if a necessary dependency is null. Add a null-check in the constructor and throw an informative RuntimeException if `map` is null (for example: `throw new IllegalArgumentException("Operation handlers map must not be null")`).